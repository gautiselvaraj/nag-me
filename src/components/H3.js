import React from 'react';
import styled from 'styled-components';

const Heading = styled.h3`
  font-size: 1.2rem;
  margin-top: 0.35rem;
  margin-bottom: 0.35rem;
`;

export default ({ children }) => <Heading>{children}</Heading>;
