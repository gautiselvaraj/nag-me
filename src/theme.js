export default {
  main:           '#214154',
  mainDarken:     '#083148',
  mainLighten:    '#215a7b',
  black:          '#111111',
  white:          '#f6f6f6',
  greyLightest:   '#efefef',
  greyLighter:    '#dddddd',
  greyLight:      '#a5a5a5',
  grey:           '#898989',
  greyDark:       '#777777',
  greyDarker:     '#595959',
  greyDarkest:    '#222',
}
