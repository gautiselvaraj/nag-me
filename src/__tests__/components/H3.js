import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import H3 from '../../components/H3';
import theme from '../../theme';
import 'jest-styled-components';

describe('<H3 />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<H3 theme={theme}>Heading 3</H3>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have correct text & element', () => {
    const h3 = mount(<H3 theme={theme}>Heading 3</H3>);
    expect(h3.text()).toBe('Heading 3');
    expect(h3.find('h3').length).toBe(1);
  });
});
