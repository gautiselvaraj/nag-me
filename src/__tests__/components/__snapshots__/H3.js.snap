// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H3 /> should match snapshot 1`] = `
.c0 {
  font-size: 1.2rem;
  margin-top: 0.35rem;
  margin-bottom: 0.35rem;
}

<h3
  className="c0"
>
  Heading 3
</h3>
`;
